mathesar_ui/src/components/abstract-type-control/AbstractTypeControl.svelte

```diff
@@ -67,8 +66,8 @@
       await save({
         type: selectedDbType,
         type_options: { ...typeOptions },
-        display_options: { ...displayOptions },
       });
```

Comment:
When a data type is changed, the default value and display options need to be reset to null. Otherwise, it will result in unexpected behaviour or type cast errors.

For eg., Take a number column, and set default value as '145'. Now change type to boolean and observe an error.

Reply:
61a725d8